quipucords/quipucords-ui
# chore(build): local run development scripts, changelog

## What's included
This PR is dependent on the Jenkinsfile updates held by #3 

* chore(build): local run development scripts, changelog
  * dev, stage, review run api scripts allowance for split repo
  * dotenv file consistency update, dotenv for test remove localhost refs
  * npm script for test dotenv fix, rename test to review, update api
  * base changelog tooling

### Notes
* Our Node development version is increasing to a minimum version of `v10.0` you'll need to update in order to help confirmation on this PR, you can use Homebrew on Mac.
* We're encouraging the use of `Yarn` over `NPM` now. Yarn can be brew installed if you're on a Mac, or even installed through NPM globally, if you already have Node installed. `$ npm install yarn -g`

## How to test
This PR updates the scripts associated with how the UI is developed. We'll need confirmation on each aspect. To start... 

1. Open your terminal, change directories into the repo, and update the NPM packages with `$ yarn`
1. Make sure you have Docker running
1. Then confirm local development with a mock api runs
    - `$ yarn start` after the compile you should be able to start viewing the app here http://localhost:3000/client/sources
    - no login screen should be available
    - you should see random data populating a GUI
    - you should be able to access a served version of the Swagger spec file, http://localhost:5050/docs/api
1. Next confirm staging development with an api runs
    - `$ yarn start:stage` after the compile you should be able to start viewing the app here https://localhost:5001/login/
    - an un-styled login screen should be available, login with the default creds
    - you should be able to edit a React component, refresh the browser manually and see the change
    - you should be able to access a served version of the Swagger spec file, http://localhost:5050/docs/api
    - **data will persist** or be saved when the container is stopped and restarted
1. And finally confirm the review environment with an api runs
    - `$ yarn start:review` after the compile you should be able to start viewing the app here open https://localhost:9443/login/
    - a styled login screen should be available, login with the default creds
    - you should be able to interact with the application as normal
    - **data will not persist** or be saved when the container is stopped and restarted










## Updates issue/story
Related to #3
